pkg/schedule/coordinator.go

```diff
@@ -152,72 +149,195 @@ func (c *Coordinator) IsPendingRegion(region uint64) bool {
 	return c.checkers.IsPendingRegion(region)
 }
 
+// PatrolRegionContext is used to store the context of patrol regions.
+type PatrolRegionContext struct {
+	// config
+	interval    time.Duration
+	workerCount int
+	scanLimit   int
+	// status
+	patrolRoundStartTime time.Time
+	durationLock         syncutil.RWMutex
+	duration             time.Duration
+	// workers
+	workersCtx    context.Context
+	workersCancel context.CancelFunc
+	regionChan    chan *core.RegionInfo
+	wg            sync.WaitGroup
+}
+
+func (p *PatrolRegionContext) init(ctx context.Context, cluster sche.ClusterInformer) {
+	p.interval = cluster.GetCheckerConfig().GetPatrolRegionInterval()
+	p.workerCount = cluster.GetCheckerConfig().GetPatrolRegionWorkerCount()
+
+	p.scanLimit = calculateScanLimit(cluster)
+	p.patrolRoundStartTime = time.Now()
+
+	p.regionChan = make(chan *core.RegionInfo, patrolRegionChanLen)
+	p.workersCtx, p.workersCancel = context.WithCancel(ctx)
+}
+
+func (p *PatrolRegionContext) stop() {
+	close(p.regionChan)
+	p.workersCancel()
+	p.wg.Wait()
+}
+
+func (p *PatrolRegionContext) updateScanLimit(cluster sche.ClusterInformer) {
+	p.scanLimit = calculateScanLimit(cluster)
```

Comment:
no need lock?

Reply:
lock is not necessary, scanLimit only be used in patrol loop.